lib/utils.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+function resolveHttpError(response, message) {
+    if (response.statusCode < 400) {
+        return;
+    }
+
+    message = message
+        ? message + " "
+        : "";
+
+    message += "Status code " + response.statusCode + " (" + response.statusMessage + ").";
+
+    if (response.body) {
+        message += " " + response.body;
+    }
+
+    var error = new Error(message);
+    error.response = response;
+
+    return error;
+}
+
+function createCallback(cb, action) {
+    return function (err, response) {
+        err = err || resolveHttpError(response, "Error " + action + ".");
+
+        if (err) {
+            return cb(err);
+        }
+
+        cb(null, response);
+    };
+}
+
+function createJsonCallback(cb, action) {
+    return function (err, response) {
+        err = err || resolveHttpError(response, "Error " + action + ".");
+
+        if (err) {
+            return cb(err);
+        }
+
+        var data = JSON.parse(response.body);
```

Comment:
Good call. If we're going to dry up JSON parsing up, it should have better error handling.